tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+use MasonUtils;
+use Spawn;
+use FileSystem;
+use TOML;
+use MasonEnv;
+use MasonNew;
+use MasonModify;
+
+/* Top Level procudeure that gets called from mason.chpl that takes in arguments from command line
+Retruns the help output in '-h' or '--help' exits in the arguements.
+If --dry-run is passed then it checks to see if the package is able to be published.
+Takes in the username of the package owner as an arguemnt*/
+proc masonPublish(args: [] string) throws {
+  try! {
+    if args.size == 2 {
+      masonPublishHelp();
+      exit(0);
+    }
+    var dry = false;
+    var username = "";
+    if args.size > 2 {
+      for arg in args[2..] {
+        if arg == '-h' || arg == '--help'{
+          masonPublishHelp();
+          exit(0);
+        }
+        else if arg == "--dry-run" {
+          dry = true;
+        }
+        else {
+          username = arg;
+        }
+      }
+      if doesGitOriginExist() {
+        if dry {
+          dryRun(username);
+        }
+        else {
+          publishPackage(username);
+        }
+      }
+      else {
+        throw new owned MasonError('Must have package set up as git repo to publish');
```

Comment:
I think this error message needs to be more specific. Consider the following use-case:

```bash
> mason new MyPackage
> cd MyPackage
> mason publish username
Must have package set up as git repo to publish
```

However, `MyPackage` _is_ set up as a git repo, just locally. I think this should notify the user that a remote origin does not exist for their repository.